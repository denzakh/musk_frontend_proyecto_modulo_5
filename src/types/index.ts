export type ItemType = {
    id: string;
    title: string;
    text: string;
    completed: boolean;
};

export type ItemTypeNullable = {
    id?: string;
    title: string;
    text: string;
    completed: boolean;
} | null;
