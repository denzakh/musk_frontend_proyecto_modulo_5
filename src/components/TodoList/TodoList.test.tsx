import { render } from '@testing-library/react';
import { TodoList } from './index';
import { ItemType } from '@/types';

export const list: ItemType[] = [
    {
        id: '1',
        title: 'title1',
        text: 'text1',
        completed: false,
    },
    { id: '2', title: 'title2', text: 'text2', completed: true },
    {
        id: '3',
        title: 'Lorem ipsum dolor sit amet consectetur adipisicing elit',
        text: 'Numquam, molestias architecto porro sit nihil sapiente alias, eligendi, ullam neque necessitatibus ut aliquid perferendis repellendus ipsum, aliquam in delectus quibusdam voluptates?',
        completed: false,
    },
];

const testFn = () => {};

test('Genera un snapshot del componente TodoItem', () => {
    const { asFragment } = render(
        <TodoList
            list={list}
            delItem={testFn}
            setCompletedItem={testFn}
            setEditedItem={testFn}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});
