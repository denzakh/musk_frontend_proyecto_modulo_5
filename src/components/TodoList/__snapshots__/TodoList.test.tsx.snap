// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genera un snapshot del componente TodoItem 1`] = `
<DocumentFragment>
  <ul
    class="list"
  >
    <li
      class="li"
      data-testid="item1"
      id="1"
      title="item"
    >
      <input
        class="input"
        data-testid="checkboxItem1"
        type="checkbox"
      />
      <div
        class="textRow"
      >
        <strong
          class="title"
        >
          title1
        </strong>
        <div
          class="text"
        >
          text1
        </div>
      </div>
      <button
        data-testid="editItem1"
        title="Edit"
        type="button"
      >
        ðŸ–‰
      </button>
      <button
        data-testid="deleteItem1"
        title="Delete"
        type="button"
      >
        ðŸ—‘
      </button>
    </li>
    <li
      class="li"
      data-testid="item3"
      id="3"
      title="item"
    >
      <input
        class="input"
        data-testid="checkboxItem3"
        type="checkbox"
      />
      <div
        class="textRow"
      >
        <strong
          class="title"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit
        </strong>
        <div
          class="text"
        >
          Numquam, molestias architecto porro sit nihil sapiente alias, eligendi, ullam neque necessitatibus ut aliquid perferendis repellendus ipsum, aliquam in delectus quibusdam voluptates?
        </div>
      </div>
      <button
        data-testid="editItem3"
        title="Edit"
        type="button"
      >
        ðŸ–‰
      </button>
      <button
        data-testid="deleteItem3"
        title="Delete"
        type="button"
      >
        ðŸ—‘
      </button>
    </li>
  </ul>
</DocumentFragment>
`;
